crates/engine/tree/src/tree/mod.rs

```diff
@@ -508,6 +510,29 @@ pub struct EngineApiTreeHandler<P, E, T: EngineTypes> {
     config: TreeConfig,
     /// Metrics for the engine api.
     metrics: EngineApiMetrics,
+    /// A bad block hook.
+    invalid_block_hook: Box<dyn InvalidBlockHook>,
```

Comment:
we could also make this optional, so that it doesn't leak into the new function and use pub set_invalid_block_hook to optional set one

but can consider doing this when we add a real impl of this

Reply:
yep, will address this in https://github.com/paradigmxyz/reth/pull/10629